packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -7,25 +7,40 @@ const FORCED_PROPS: Partial<GridComponentProps> = {
   pagination: true,
   disableMultipleColumnsSorting: true,
   disableMultipleSelection: true,
+  disableColumnResize: true,
 };
 
 export type DataGridProps = Omit<
   GridComponentProps,
   | 'disableMultipleColumnsSorting'
   | 'disableMultipleSelection'
+  | 'disableColumnResize'
   | 'licenseStatus'
   | 'options'
   | 'pagination'
 > & {
   disableMultipleColumnsSorting?: true;
   disableMultipleSelection?: true;
+  disableColumnResize?: true;
   pagination?: true;
 };
 
 const MAX_PAGE_SIZE = 100;
 
 const DataGrid2 = React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(props, ref) {
-  const { className, pageSize: pageSizeProp, ...other } = props;
+  const { className, pageSize: pageSizeProp, columns, ...other } = props;
+
+  const hasResizableColumns = columns.some((c) => c.resizable);
+  if (hasResizableColumns) {
+    throw new Error(
```

Comment:
What about using a custom prop types? This would allow the code to be removed in product (no bundle size implication). This would also be less disruptive to the DX.

Reply:
> something wrong in the configuration of tsconfig?

Probably due to all the paths in tsconfig, as the npm pkgs are missing